src/dxtbx/format/FormatNXmx.py

```diff
@@ -7,6 +7,21 @@
 from dxtbx.format.FormatNexus import FormatNexus
 
 
+# A singleton to hold unique static_mask objects to avoid duplications
+class MaskDict(dict):
+    def hash_func(self, mask):
+        return hash(mask[0].as_numpy_array().tobytes())
```

Comment:
if the mask tuple has length > 1 we would also want to include the other elements in the hash, or have the key be a tuple of the hash of each panel mask: I appreciate for the problem you are trying to solve here the length of this tuple is exactly 1